compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELinalgFunctionOutlining.cpp

```diff
@@ -20,48 +20,29 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
-/// Return true if the strides of `type` make it a contiguous memref.
-bool isContiguousMemRef(MemRefType type) {
-  ArrayRef<int64_t> shape = type.getShape();
-  SmallVector<int64_t, 4> strides;
-  int64_t ignoredOffset;
-  if (failed(getStridesAndOffset(type, strides, ignoredOffset))) return false;
-  int64_t expectedStride = 1;
-  for (int i = shape.size() - 1; i >= 0; --i) {
-    if (shape[i] == ShapedType::kDynamic) return false;
-    if (strides[i] != expectedStride) return false;
-    expectedStride *= shape[i];
-  }
-  return true;
-}
-
-/// If `type` is a contiguous memref, return an equivalent memref without any
-/// layout attribute. Otherwise, return nullptr.
-Type getTypeWithoutLayout(Type type) {
+/// Return true if `type` is a memref with an identity layout.
+bool isMemRefWithIdentityLayout(Type type) {
   auto memRefType = dyn_cast<MemRefType>(type);
-  if (!memRefType) return {};
-  if (!isContiguousMemRef(memRefType)) return {};
-  return MemRefType::get(memRefType.getShape(), memRefType.getElementType(),
-                         MemRefLayoutAttrInterface{},
-                         memRefType.getMemorySpace());
+  if (!memRefType) return false;
```

Comment:
I think it should throw an error instead of return false.

Reply:
I guess I could refactor this function to take a memref argument.